pinot-controller/src/main/java/org/apache/pinot/controller/validation/RealtimeSegmentValidationManager.java

```diff
@@ -67,6 +67,22 @@ public RealtimeSegmentValidationManager(ControllerConf config, PinotHelixResourc
     Preconditions.checkState(_segmentLevelValidationIntervalInSeconds > 0);
   }
 
+  @Override
+  protected void setUpTask() {
+    // Prefetch the LLC segment without segment store copy from ZK, which helps to alleviate ZK access.
```

Comment:
Are you sure that the controller leadership for a table is established by this time? We should do this in a callback from LeadControllerManager. And yes, this needs to be coded up. There is no callback right now, afaik.

@jackjlli  can help.

Reply:
What does the callback do exactly? In `setUpPinotController()` of the ControllerStart, the controllerPeriodicTask was setup after the `LeadControllerManager` init and the `registerAndConnectAsHelixParticipant()`. In `registerAndConnectAsHelixParticipant`, the table leadership is determined via Helix transition. Are you suggesting the Helix transition happens asynchronously? @mcvsubbu

Reply:
Yes, all helix transitions happen asynchronously.
The controllers will get state transitions -- one for each partition of the `leadControllerResource`, and based on that can decide whether a certain table is to be processed on that controller or not.

Reply:
Added a TODO for this. Not planning to include the change in this pr.

Reply:
Where is the TODO? Please add it here, otherwise someone going through the code will have the same question (worse, may make the same assumption in other places). Make it clear that this may not work all the time and it is a TODO to take action when leadership is established.

Reply:
TODO is above this function: [line 71](https://github.com/apache/incubator-pinot/pull/6778/files#diff-0a9ffdab99ed314e7a8e58bf86c125f020b7d6f86493d1f6d719beab62b7f106R71).